docs/source/contributing/contributing-code.rst

```diff
@@ -0,0 +1,90 @@
+Contributing code
+==================================================
+
+Local setup
+----------------------------------
+
+To contribute code, basic knowledge of git is required, if you are unfamiliar
+with the git workflow, checkout `GitHub's tutorial <https://docs.github.com/en/get-started/quickstart/hello-world>`_.
+The following steps allow you to contribute code to DoWhy.
+
+1. Fork the `dowhy main repository <https://github.com/py-why/dowhy>`_
+2. Clone this repository to your local machine using
+
+.. code:: shell
+
+   git clone https://github.com/<YOUR_GITHUB_USERNAME>/dowhy
+
+3. [optional] Create a virtual environment using python venv or conda #check if poetry does this automatically
+
+.. code:: shell
+
+   virtualenv dowhy-env --python=python3.8
```

Comment:
If I recall correctly, I believe the preferred way [has been changed](https://docs.python.org/3/library/venv.html) to "python -m venv" in python 3+. Worth checking out, I'm not sure.

Reply:
It's probably worth removing this step and replacing it with a note that poetry will install a virtualenv.

Reply:
Well, I never use that. I prefer to manage my environments more directly myself. Not sure if I'm a total exception in this workflow though : ). However, given that creating and managing virtual environments is well documented elsewhere and nothing special needs to happen here, I agree with removing it.